src/hock/QueryStringHock.jsx

```diff
@@ -152,25 +166,43 @@ const QueryStringHock = (config: ?Object = null, onQueryChangeFunction: ?Functio
         }
 
         QueryStringHock.propTypes = {
-            location: PropTypes.object.isRequired, // must be react router location object, required for react-router v1 and v2
-            history: PropTypes.object // must be react router history object, required for react-router v1
+            location: PropTypes.object.isRequired,
+            history: PropTypes.object
         };
 
         QueryStringHock.contextTypes = {
-            router: React.PropTypes.object // required for react-router v2
+            router: React.PropTypes.object
         };
 
         return QueryStringHock;
     }
 };
 
-export default QueryStringHock;
+export default QueryStringDecorator;
+
+
+/**
+ * Configuration object for the QueryStringDecorator.
+ *
+ * @typedef QueryStringDecoratorConfig
+ * @property {Object} [defaultQuery] These defaults will be passed down in the query prop whenever they aren't present in the actual query string.
+ * @property {string} [queryPropName = "query"] Sets the name of the query prop.
+ * @property {boolean} [replaceState = false] Optional boolean, setting this to true will make query changes use replaceState instead of pushState
+ * @property {Array<string>} [arrayParams] If you have particular query parameters that you always want to return as arrays, name them in here. All arrayParams will also be an empty array if they are not present in the query string. By default react-router only passes an array of query param values back if there are more than one value in them.
+ */
+
+/**
+ * A decorator function that accepts a component to decorate, and returns that decorated component.
+ * @callback QueryStringApplier
+ * @param {ReactComponent} ComponentToDecorate The component you wish to wrap in a `QueryStringHock`.
+ * @return {ReactComponent} The decorated component.
+ */
 
 /**
- * A function to be called when the query string has changed.
+ * A function to be called when the query string changes. By default this function is called on initial `componentDidMount` and every time any query param changes after this - to limit this use the third argument `onChangeParams`
```

Comment:
Same line length stuff for this file, probably need to enforce it with eslint but for the moment I think sub 100 char is a good marker.